system/jlib/jsecrets.cpp

```diff
@@ -259,6 +260,10 @@ class CVault
         name.set(vault->queryProp("@name"));
         kind = getSecretType(vault->queryProp("@kind"));
 
+        vaultNamespace.set(vault->queryProp("@namespace"));
+        if (vaultNamespace.length())
+            addPathSepChar(vaultNamespace, '/');
```

Comment:
@afishbeck is the trailing / correct?  All the examples I have seen do not have a trailing / e.g.

```
--header "X-Vault-Namespace: some/path
X-Vault-Namespace: root
```

Reply:
It could be optional.  Vault cli has an option to generate a CURL example for each command (-output-curl-string).  The CURL example used a forward slash at the end of the namespace.  We tested via curl with the forward slash.  I don't remember if we also tried without it to see if it would work, but I decided for now at least to error on the side of matching what vault cli suggested.
I can't test namespaces on my non-enterprise vault but I could work with Fernando again to see if we can remove the slash.  Or I could open a JIRA to try removing the slash next time we are working with the enterprise vault.